src/basic/parse-util.c

```diff
@@ -162,15 +165,15 @@ int parse_size(const char *t, uint64_t base, uint64_t *size) {
                 unsigned i;
 
                 p += strspn(p, WHITESPACE);
-                if (*p == '-')
-                        return -ERANGE;
 
                 errno = 0;
                 l = strtoull(p, &e, 10);
-                if (errno > 0)
+                if (errno != 0)
```

Comment:
I see what you're doing here... but I find that this looks slightly odd:

```
  if (errno != 0)
    return -errno;
```

So if `errno` is ever negative you'll end up returning a postive value?

Perhaps this would be slightly better?

```
  if (errno != 0)
    return -abs(errno);
```